### PR TITLE
Promote Graceful termination to stable

### LINKED ISSUE #4611: TEP-0058: Graceful Termination - Promotions and Deprecation
In [TEP-0058: Graceful Termination](https://github.com/tektoncd/community/blob/main/teps/0058-graceful-pipeline-run-termination.md), we deprecated `PipelineRunCancelled` and added:
- `StoppedRunFinally` - To "stop" (i.e. let the tasks complete, then execute finally tasks) a Pipeline
- `CancelledRunFinally` - To "cancel" (i.e. interrupt any executing non finally tasks, then execute finally tasks)
- `Cancelled` - Same as today's `PipelineRunCancelled` - i.e. interrupt any executing tasks without running finally tasks

The above was released in [v0.25](https://github.com/tektoncd/pipeline/releases/tag/v0.25.0) in June 2021. 

The new fields are still in [alpha](https://github.com/tektoncd/pipeline/blob/302895b5a1ca45c02a85a9822201643c159fe02c/docs/install.md#alpha-features), so this issue tracks promoting them to beta. This issue also tracks deprecation and removal of `PipelineRunCancelled`.

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

With the addition of graceful termination in v0.25.x release,
`PipelineRunCancelled` was deprecated. It's been 8 months and no such
issues so far, we can promote these to beta.    

In this patch, unguarding the graceful termination from alpha feature
flag and modifying the tests based on this.

closes #4611 

Signed-off-by: vinamra28 <jvinamra776@gmail.com>

/kind feature

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [X] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [X] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [X] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes
```release-note
Promoting graceful termination of pipelinerun to beta and keeping `PipelineRunCancelled` as deprecated, it will be removed after 3 releases
```

### CODE REVIEW

File: pkg/apis/pipeline/v1beta1/pipelinerun_types.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (pr *PipelineRun) PipelineTimeout(ctx context.Context) time.Duration {
 	if pr.Spec.Timeouts != nil && pr.Spec.Timeouts.Pipeline != nil {
 		return pr.Spec.Timeouts.Pipeline.Duration
 	}
-	return time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes) * time.Minute
+	return time.Duration(apisconfig.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes) * time.Minute
```
Comment (jerop) [inline pkg/apis/pipeline/v1beta1/pipelinerun_types.go:105]: looks like this change in `PipelineTimeout` is because of the refactor removing the duplicated import, can we move this cleanup to its own PR?

**Review by jerop — COMMENTED** [review]
thanks for helping out here @vinamra28, just look a quick glance but will wait for it to be ready for review 

/assign 

closes https://github.com/tektoncd/pipeline/issues/4611

File: pkg/apis/pipeline/v1beta1/pipelinerun_types.go (hunk 1 above)
Comment (vinamra28) [reply pkg/apis/pipeline/v1beta1/pipelinerun_types.go:105]: @jerop thanks for the review, yeah I'll revert this

File: test/v1alpha1/cancel_test.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ spec:
 			}
 
 			t.Logf("Waiting for PipelineRun %s in namespace %s to be cancelled", pipelineRunName, namespace)
-			if err := WaitForPipelineRunState(ctx, c, pipelineRunName, pipelineRunTimeout, FailedWithReason("PipelineRunCancelled", pipelineRunName), "PipelineRunCancelled"); err != nil {
+			if err := WaitForPipelineRunState(ctx, c, pipelineRunName, pipelineRunTimeout, FailedWithReason("Cancelled", pipelineRunName), "PipelineRunCancelled"); err != nil {
```
Comment (jerop) [inline test/v1alpha1/cancel_test.go:118]: I believe v1alpha1 should continue using `PipelineRunCancelled` 🤔 

(v1alpha1 is read-only, we don't make change to it)

**Review by jerop — COMMENTED** [review]
thanks again @vinamra28! 

please remove graceful termination from the alpha features in the install docs - https://github.com/tektoncd/pipeline/blob/f2da9a877f295a6cb865d8dee7b52872089c061d/docs/install.md#alpha-features

**Comment by jerop** [comment]
@vinamra28 - we should wait for the fixes related to graceful stopping as described in https://github.com/tektoncd/pipeline/pull/4651#issuecomment-1068147952 to go in first

**Comment by vdemeester** [comment]
@jerop @vinamra28 For this, I think we should unguard the new values *but* keep the old value (`Cancelled`) for a while, and deprecate it. By doing tihs, we ensure we give more time to user that never enabled the feature, through feature-flags, to keep using the old value (and behavior) for a little while.

Does it make sense ?
cc @abayer @imjasonh

File: pkg/pipelinerunmetrics/metrics.go [hunk 3]
```diff
@@ -63,8 +63,6 @@ var (
 const (
 	// ReasonCancelled indicates that a PipelineRun was cancelled.
 	ReasonCancelled = "Cancelled"
-	// ReasonCancelledDeprecated Deprecated: "PipelineRunCancelled" indicates that a PipelineRun was cancelled.
-	ReasonCancelledDeprecated = "PipelineRunCancelled"
 )
```
Comment (vinamra28) [inline pkg/pipelinerunmetrics/metrics.go:66]: @vdemeester @jerop @imjasonh shall I revert this removal of deprecated reason or is this fine?

Comment (vdemeester) [reply pkg/pipelinerunmetrics/metrics.go:66]: In my opinion, we should keep `PipelineRunSpecStatusCancelledDeprecated` (and thus from user's perspective `PipelineRunCancelled`) for this release, allowing to use `Cancelled` as well as `PipelineRunCancelled` for a time being.
And we should add a deprecation notice on `PipelineRunSpecStatusCancelledDeprecated` (and thus from the user's perspective `PipelineRunCancelled`) that it will be removed in X releases (3 at least)

Comment (vdemeester) [reply pkg/pipelinerunmetrics/metrics.go:66]: for the deprecated reason, I think we should keep it as long as we keep the deprecated field value as well.

**Comment by vinamra28** [comment]
@vdemeester @jerop @pritidesai @imjasonh I have updated the PR based on above feedbacks. PTAL

File: test/v1alpha1/cancel_test.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ import (
 
 	"github.com/tektoncd/pipeline/test/parse"
 
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
```
Comment (lbernick) [inline test/v1alpha1/cancel_test.go:32]: looks like this test is meant for v1alpha1, maybe keep this as is?

File: pkg/apis/pipeline/v1beta1/pipelinerun_validation.go [hunk 5]
```diff
@@ -129,20 +129,15 @@ func validateSpecStatus(ctx context.Context, status PipelineRunSpecStatus) *apis
...
 
-	cfg := config.FromContextOrDefaults(ctx)
-	if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
-		return apis.ErrInvalidValue(fmt.Sprintf("%s should be %s, %s, %s or %s", status,
-			PipelineRunSpecStatusCancelled,
-			PipelineRunSpecStatusCancelledRunFinally,
-			PipelineRunSpecStatusStoppedRunFinally,
-			PipelineRunSpecStatusPending), "status")
-	}
-	return apis.ErrInvalidValue(fmt.Sprintf("%s should be %s or %s", status,
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/pipelinerun_validation.go:143]: It seems like removing this will mean that the deprecated value will no longer work, and it seems like from previous PR comments that we want to avoid removing this value until the new value is promoted to beta.

Comment (vinamra28) [reply pkg/apis/pipeline/v1beta1/pipelinerun_validation.go:143]: @lbernick it's just the webhook validation which was checking for the fields that were behind feature flag.

File: test/v1alpha1/cancel_test.go (hunk 4 above)
Comment (vinamra28) [reply test/v1alpha1/cancel_test.go:32]: yeah test is meant for v1alpha1 but the status being used here were from v1beta1. Once `PipelineRunCancelled` is removed, we anyhow will have to use v1beta1 status. Moreover v1alpha1 gets converted to v1beta1 during runtime

**Review by lbernick — COMMENTED** [review]
/lgtm

File: pkg/apis/pipeline/v1beta1/pipelinerun_validation.go (hunk 5 above)
Comment (vdemeester) [reply pkg/apis/pipeline/v1beta1/pipelinerun_validation.go:143]: @lbernick it's just the error message. We could still include `PipelineRunSpecStatusCancelledDeprecated` but as it is deprecated, I don't really think it would make sense.

**Review by vdemeester — APPROVED** [review]
We should add a deprecated note for removing `PipelineRunSpecStatusCancelledDeprecated` though, in the doc

**Comment by lbernick** [comment]
> We should add a deprecated note for removing `PipelineRunSpecStatusCancelledDeprecated` though, in the doc

@vdemeester are you referring to [this file](https://github.com/tektoncd/pipeline/blob/main/docs/deprecations.md)? it's already there
